x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -235,23 +238,26 @@ protected void masterOperation(
             return;
         }
 
+        ActionListener<Boolean> finishedStoringListener = ActionListener.wrap(bool -> {
+            TrainedModelConfig configToReturn = trainedModelConfig.clearDefinition().build();
+            if (modelPackageConfigHolder.get() != null) {
+                triggerModelFetchIfNecessary(
+                    configToReturn.getModelId(),
+                    modelPackageConfigHolder.get(),
+                    request.isWaitForCompletion(),
+                    ActionListener.wrap(
+                        downloadTriggered -> listener.onResponse(new PutTrainedModelAction.Response(configToReturn)),
+                        listener::onFailure
+                    )
+                );
+            } else {
+                listener.onResponse(new PutTrainedModelAction.Response(configToReturn));
+            }
+        }, listener::onFailure);
+
+        var isPackageModel = config.isPackagedModel();
```

Comment:
We need to leverage `isPackageModel` to determine if we can overwrite when we call `storeTrainedModel` below